File: app/src/main/java/com/adiaz/deportesmadrid/fragments/TabTeamCalendar.java
package com.adiaz.deportesmadrid.fragments;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.adiaz.deportesmadrid.R;
import com.adiaz.deportesmadrid.adapters.TeamMatchesAdapter;
import com.adiaz.deportesmadrid.callbacks.CompetitionCallback;
import com.adiaz.deportesmadrid.retrofit.groupsdetails.MatchRetrofit;
import com.adiaz.deportesmadrid.utils.Constants;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class TabTeamCalendar extends Fragment {



    @BindView(R.id.rv_team_matches)
    RecyclerView rvTeamMatches;

    CompetitionCallback mCompetitionCallback;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try {
            mCompetitionCallback = (CompetitionCallback) context;
        } catch (ClassCastException e) {
            throw new ClassCastException(context.toString() + " must implement CompetitionCallback");
        }
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_team_calendar, container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        List<MatchRetrofit> teamMatches = mCompetitionCallback.queryMatchesList();
        if (teamMatches!=null) {
            LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this.getContext());
            String teamName = mCompetitionCallback.queryTeam()==null ? Constants.FIELD_EMPTY : mCompetitionCallback.queryTeam().getName();
            TeamMatchesAdapter teamMatchesAdapter = new TeamMatchesAdapter(this.getContext(), mCompetitionCallback.queryMatchesList(), teamName);
            rvTeamMatches.setHasFixedSize(true);
            rvTeamMatches.setLayoutManager(linearLayoutManager);
            rvTeamMatches.setAdapter(teamMatchesAdapter);
            teamMatchesAdapter.notifyDataSetChanged();
        }
    }
}

File: app/src/main/java/com/adiaz/ligasmadrid/fragments/TabTeamCalendar.kt
package com.adiaz.ligasmadrid.fragments

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.adiaz.ligasmadrid.R
import com.adiaz.ligasmadrid.adapters.TeamMatchesAdapter
import com.adiaz.ligasmadrid.callbacks.CompetitionCallback
import com.adiaz.ligasmadrid.utils.Constants
import kotlinx.android.synthetic.main.fragment_team_calendar.*

class TabTeamCalendar : Fragment() {

    lateinit var mCompetitionCallback: CompetitionCallback
    lateinit var mListItemClickListener: TeamMatchesAdapter.ListItemClickListener

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        try {
            mCompetitionCallback = context as CompetitionCallback
            mListItemClickListener = context as TeamMatchesAdapter.ListItemClickListener
        } catch (e: ClassCastException) {
            throw ClassCastException(context!!.toString() + " must implement CompetitionCallback")
        }

    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater!!.inflate(R.layout.fragment_team_calendar, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        val teamMatches = mCompetitionCallback.queryMatchesList()
        if (teamMatches != null) {
            val linearLayoutManager = LinearLayoutManager(this.context)
            val teamName = if (mCompetitionCallback.queryTeam() == null) Constants.FIELD_EMPTY else mCompetitionCallback.queryTeam()!!.name
            val teamMatchesAdapter = TeamMatchesAdapter(this.context, mCompetitionCallback.queryMatchesList()!!, teamName, mListItemClickListener)
            rvTeamMatches.setHasFixedSize(true)
            rvTeamMatches.layoutManager = linearLayoutManager
            rvTeamMatches.adapter = teamMatchesAdapter
            teamMatchesAdapter.notifyDataSetChanged()
        }
    }
}
